.github/workflows/test.yml

```diff
@@ -1,323 +1,85 @@
+                         ./tests/end_to_end/http_server/http_outcall_fetch
+                         ./tests/end_to_end/http_server/ic_evm_rpc
+                         ./tests/property/candid_rpc/class_api/stable_b_tree_map
+                         ./tests/property/candid_rpc/functional_api/stable_b_tree_map
+                         ') || '' }}
+
+    determine-should-run-tests:
+        name: Determine if tests should run
+        runs-on: ubuntu-latest
+        outputs:
+            should_run_tests: ${{ steps.determine_should_run_tests.outputs.should_run_tests }}
+        steps:
+            - uses: actions/checkout@v4
+
+            - id: determine_should_run_tests
+              # Determine if tests should run for this pr. If it's a manually submitted release candidate no tests will run.
               run: |
                   BRANCH_NAME="${{ github.head_ref }}"
                   RELEASE_VERSION="${BRANCH_NAME:9}"
                   COMMIT_MESSAGE=$(git log -1 --pretty=format:"%s")
-                  if [[ "${{ contains(github.head_ref, 'release--') }}" == "true" && "$COMMIT_MESSAGE" != "azle-bot automated release $RELEASE_VERSION" ]]
-                  then
-                    ./publish-github-action.sh $RELEASE_VERSION ${{ toJSON(steps.example_directories.outputs.example_directories) }}
+
+                  IS_AUTOMATED_RELEASE="false"
+                  if [[ "$COMMIT_MESSAGE" == "azle-bot automated release $RELEASE_VERSION" ]]; then
+                      IS_AUTOMATED_RELEASE="true"
+                  fi
+
+                  if [[ "${{env.IS_RELEASE_CANDIDATE}}" == "true" && "$IS_AUTOMATED_RELEASE" == "false" ]]; then
+                      echo "should_run_tests=false" >> "$GITHUB_OUTPUT"
                   else
-                    echo "::set-output name=should_run_tests::true"
+                      echo "should_run_tests=true" >> "$GITHUB_OUTPUT"
                   fi
-    basic-integration-tests:
-        needs: release-candidate-deploy
+
+    run-test:
+        # Tests can either run against the current code of Azle found in the repository, or the code deployed by the GitHub Action to npm.
+        # Feature branch pull requests (pull requests without release-- in the base branch name) will run all tests against the code found in the repository.
+        # Release branch pull requests (pull requests with release-- in the base branch name) will run all tests against the code found in the repository and the code deployed by the GitHub Action to npm.
+        # Pushes to main will run all tests against the code in the repository if the latest commit was not a merge of a release branch, and will run tests against the code in the repository and the code deployed by the GitHub Action to npm otherwise.
+        name: '${{matrix.tests.name}} | ${{matrix.tests.type}} | ${{matrix.tests.syntax}} | ${{matrix.tests.api}} | ${{matrix.azle_source}}'
+        needs:
+            - determine-should-run-tests
```

Comment:
Can't should run tests be determined by the build workflow? Can we connect them somehow? Can we determine should release and should run tests at the same time?

Reply:
yeah we could put it into an action

done